src/MolSim.cpp

```diff
@@ -7,38 +7,55 @@
+    double end_time;
+    double delta_t;
+
+    boost::program_options::options_description options_desc("Allowed options");
+    options_desc.add_options()("help,h", "produce help message");
+    options_desc.add_options()("input_file_path,f", boost::program_options::value<std::string>(&input_file_path),
+                               "The path to the input file. Must be specified, otherwise the program will terminate. Can be inserted as positional argument.");
+    options_desc.add_options()("output_dir_path,o", boost::program_options::value<std::string>(&output_dir_path),
+                               "The path to the directory in which to save the simulation output files. If not specified, the output will be saved in the programs working directory."
+                               "Note: If the directory at the specified path does not exist, it will be created.");
+    options_desc.add_options()("delta_t,d", boost::program_options::value<double>(&delta_t)->default_value(0.014),
+                               "The time step per simulation iteration");
+    options_desc.add_options()("end_time,e", boost::program_options::value<double>(&end_time)->default_value(1000),
+                               "The time, at which the simulation will end");
+
+    boost::program_options::positional_options_description positional_options_desc;
+    positional_options_desc.add("input_file_path", -1);
+
+    boost::program_options::variables_map variables_map;
+    boost::program_options::store(boost::program_options::command_line_parser(argc, argsv).options(options_desc).positional(positional_options_desc).run(), variables_map);
+    boost::program_options::notify(variables_map);
+
+    if (argc <= 1 || variables_map.count("help")) {
+        std::cout << options_desc << std::endl;
+        exit(-1);
     }
-
-    if (!vm.count("input_file_path")) {
+    if (!variables_map.count("input_file_path")) {
         std::cout << "Error: no input file path given." << std::endl;
-        std::cout << desc << std::endl;
-        return -1;
+        std::cout << options_desc << std::endl;
+        exit(-1);
+    }
+    if (delta_t <= 0) {
+        std::cout << "Error: timestep delta_t must be greater than 0." << std::endl;
+        exit(-1);
     }
+    if (end_time <= 0) {
+        std::cout << "Error: end time must be greater than 0." << std::endl;
+        exit(-1);
+    }
+
+    std::cout << "Configuring simulation with the following parameters:" << std::endl;
+    std::cout << "Input file path: " << input_file_path << std::endl;
+    std::cout << "Output directory path: " << output_dir_path << std::endl;
+    std::cout << "Delta t: " << delta_t << std::endl;
+    std::cout << "End time: " << end_time << std::endl;
 
-    Simulation simulation{inputFilepath, Simulation::IntegrationMethod::VERLET, deltaT, endTime};
+    Simulation simulation{input_file_path, output_dir_path, Simulation::IntegrationMethod::VERLET, delta_t, end_time};
```

Comment:
Passing the input file path is bad, because we may have dynamic data. (E.g spawing particles in a rectangle)

Maybe an Input-Wrapper which could select between `DYNAMIC` and `STATIC` data. Dynamic data is generated on the fly (like as a rectangle) and static data can be read from the filesystem.

The wrapper should implement a function `(DATATYPE, file_path / particle_generator) ==> ParticleContainer`  that creates the needed initial conditions

Reply:
We should still use only file input, as the amount of arguments necessary to generate a cuboid can't be comfortably passed via the commandline. It's better to use a custom file format and keep the input file method.